File: Views/Subscription/SubscriptionInheritedView.swift
import SwiftUI

struct SubscriptionInheritedView: View {
    let partnerName: String
    let onContinue: () -> Void
    @State private var showAnimation = false
    
    var body: some View {
        ZStack {
            // Fond gradient premium
            LinearGradient(
                gradient: Gradient(colors: [
                    Color(hex: "#FFD700"),
                    Color(hex: "#FFA500"),
                    Color(hex: "#FF8C00")
                ]),
                startPoint: .topLeading,
                endPoint: .bottomTrailing
            )
            .ignoresSafeArea()
            
            VStack(spacing: 40) {
                Spacer()
                
                // Animation de couronne premium
                VStack(spacing: 20) {
                    // Icône de couronne animée
                    ZStack {
                        Circle()
                            .fill(Color.white.opacity(0.2))
                            .frame(width: 120, height: 120)
                            .scaleEffect(showAnimation ? 1.2 : 1.0)
                            .opacity(showAnimation ? 0.3 : 0.8)
                            .animation(
                                Animation.easeInOut(duration: 2.0)
                                    .repeatForever(autoreverses: true),
                                value: showAnimation
                            )
                        
                        Image(systemName: "crown.fill")
                            .font(.system(size: 50))
                            .foregroundColor(.white)
                            .scaleEffect(showAnimation ? 1.1 : 1.0)
                            .animation(
                                Animation.easeInOut(duration: 1.5)
                                    .repeatForever(autoreverses: true),
                                value: showAnimation
                            )
                    }
                    
                    // Titre premium
                    Text("premium_unlocked".localized)
                        .font(.system(size: 28, weight: .bold))
                        .foregroundColor(.white)
                        .multilineTextAlignment(.center)
                        .padding(.horizontal, 30)
                    
                    // Message avec nom du partenaire
                    Text(String(format: "premium_shared_message".localized, partnerName))
                        .font(.system(size: 18))
                        .foregroundColor(.white.opacity(0.9))
                        .multilineTextAlignment(.center)
                        .padding(.horizontal, 30)
                    
                    // Liste des avantages
                    VStack(spacing: 12) {
                                            premiumFeatureRow(icon: "🔓", text: "premium_features_unlocked".localized)
                    premiumFeatureRow(icon: "🔥", text: "unlimited_questions".localized)
                    premiumFeatureRow(icon: "💎", text: "exclusive_premium_content".localized)
                    }
                    .opacity(showAnimation ? 1.0 : 0.0)
                    .animation(.easeInOut(duration: 1.0).delay(1.5), value: showAnimation)
                }
                
                Spacer()
                
                // Bouton Continuer
                Button(action: {
                    print("🎁 SubscriptionInheritedView: Bouton Continuer pressé")
                    onContinue()
                }) {
                    Text("discover_premium".localized)
                        .font(.system(size: 18, weight: .bold))
                        .foregroundColor(.white)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, 18)
                        .background(Color.white)
                        .cornerRadius(25)
                }
                .padding(.horizontal, 30)
                .padding(.bottom, 50)
                .opacity(showAnimation ? 1.0 : 0.0)
                .animation(.easeInOut(duration: 1.0).delay(2.0), value: showAnimation)
            }
        }
        .onAppear {
            print("🎁 SubscriptionInheritedView: Vue apparue pour héritage de: \(partnerName)")
            showAnimation = true
        }
    }
    
    private func premiumFeatureRow(icon: String, text: String) -> some View {
        HStack(spacing: 15) {
            Text(icon)
                .font(.system(size: 24))
            
            Text(text)
                .font(.system(size: 16, weight: .medium))
                .foregroundColor(.white)
            
            Spacer()
        }
        .padding(.horizontal, 40)
    }
} 

File: Views/Subscription/SubscriptionInheritedView.kt
// SubscriptionInheritedView.kt
// Remplace le package par le tien
package com.yourapp.ui.subscription

import android.util.Log
import androidx.compose.animation.AnimatedVisibility
import androidx.compose.animation.core.FastOutSlowInEasing
import androidx.compose.animation.core.RepeatMode
import androidx.compose.animation.core.animateFloat
import androidx.compose.animation.core.infiniteRepeatable
import androidx.compose.animation.core.rememberInfiniteTransition
import androidx.compose.animation.core.tween
import androidx.compose.animation.fadeIn
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.rounded.WorkspacePremium
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.graphicsLayer
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.yourapp.R // ⟵ remplace par le nom de package R de ton app

@Composable
fun SubscriptionInheritedView(
    partnerName: String,
    onContinue: () -> Unit
) {
    // Dégradé premium (or → orange)
    val premiumGradient = Brush.linearGradient(
        colors = listOf(
            Color(0xFFFFD700), // #FFD700
            Color(0xFFFFA500), // #FFA500
            Color(0xFFFF8C00)  // #FF8C00
        )
    )

    // Animations “pulse” (cercle et icône)
    val infinite = rememberInfiniteTransition(label = "crownPulse")
    val circleScale by infinite.animateFloat(
        initialValue = 1.0f,
        targetValue = 1.2f,
        animationSpec = infiniteRepeatable(
            animation = tween(durationMillis = 2000, easing = FastOutSlowInEasing),
            repeatMode = RepeatMode.Reverse
        ),
        label = "circleScale"
    )
    val circleAlpha by infinite.animateFloat(
        initialValue = 0.8f,
        targetValue = 0.3f,
        animationSpec = infiniteRepeatable(
            animation = tween(durationMillis = 2000),
            repeatMode = RepeatMode.Reverse
        ),
        label = "circleAlpha"
    )
    val iconScale by infinite.animateFloat(
        initialValue = 1.0f,
        targetValue = 1.1f,
        animationSpec = infiniteRepeatable(
            animation = tween(durationMillis = 1500, easing = FastOutSlowInEasing),
            repeatMode = RepeatMode.Reverse
        ),
        label = "iconScale"
    )

    // Logging équivalent au onAppear Swift
    LaunchedEffect(partnerName) {
        Log.d("SubscriptionInheritedView", "Vue apparue pour héritage de: $partnerName")
    }

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(premiumGradient)
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(horizontal = 30.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Spacer(Modifier.weight(1f))

            // Bloc “couronne” + titres
            Column(horizontalAlignment = Alignment.CenterHorizontally, verticalArrangement = Arrangement.spacedBy(20.dp)) {
                Box(
                    contentAlignment = Alignment.Center,
                    modifier = Modifier.size(120.dp)
                ) {
                    // Cercle pulsant
                    Box(
                        modifier = Modifier
                            .matchParentSize()
                            .graphicsLayer(
                                scaleX = circleScale,
                                scaleY = circleScale,
                                alpha = circleAlpha
                            )
                            .background(Color.White.copy(alpha = 0.2f), CircleShape)
                    )
                    // Icône de “couronne”
                    Icon(
                        imageVector = Icons.Rounded.WorkspacePremium,
                        contentDescription = null,
                        tint = Color.White,
                        modifier = Modifier
                            .size(50.dp)
                            .graphicsLayer(
                                scaleX = iconScale,
                                scaleY = iconScale
                            )
                    )
                }

                // Titre premium
                Text(
                    text = stringResource(R.string.premium_unlocked),
                    fontSize = 28.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.White,
                    textAlign = TextAlign.Center,
                    modifier = Modifier.padding(horizontal = 30.dp)
                )

                // Message avec nom du partenaire
                Text(
                    text = stringResource(R.string.premium_shared_message, partnerName),
                    fontSize = 18.sp,
                    color = Color.White.copy(alpha = 0.9f),
                    textAlign = TextAlign.Center,
                    modifier = Modifier.padding(horizontal = 30.dp)
                )

                // Liste des avantages (apparition retardée comme dans Swift)
                AnimatedVisibility(
                    visible = true,
                    enter = fadeIn(animationSpec = tween(durationMillis = 1000, delayMillis = 1500))
                ) {
                    Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
                        PremiumFeatureRow(icon = "🔓", text = stringResource(R.string.premium_features_unlocked))
                        PremiumFeatureRow(icon = "🔥", text = stringResource(R.string.unlimited_questions))
                        PremiumFeatureRow(icon = "💎", text = stringResource(R.string.exclusive_premium_content))
                    }
                }
            }

            Spacer(Modifier.weight(1f))

            // Bouton Continuer (apparition retardée)
            AnimatedVisibility(
                visible = true,
                enter = fadeIn(animationSpec = tween(durationMillis = 1000, delayMillis = 2000))
            ) {
                Button(
                    onClick = {
                        Log.d("SubscriptionInheritedView", "🎁 Bouton Continuer pressé")
                        onContinue()
                    },
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(bottom = 50.dp),
                    shape = MaterialTheme.shapes.large, // arrondi généreux
                    colors = ButtonDefaults.buttonColors(
                        containerColor = Color.White,
                        // Lisibilité: texte “or” sur fond blanc
                        contentColor = Color(0xFFFF8C00)
                    )
                ) {
                    Text(
                        text = stringResource(R.string.discover_premium),
                        fontSize = 18.sp,
                        fontWeight = FontWeight.Bold,
                        modifier = Modifier.padding(vertical = 6.dp)
                    )
                }
            }
        }
    }
}

@Composable
private fun PremiumFeatureRow(icon: String, text: String) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 40.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Text(text = icon, fontSize = 24.sp)
        Spacer(Modifier.width(15.dp))
        Text(
            text = text,
            fontSize = 16.sp,
            fontWeight = FontWeight.Medium,
            color = Color.White
        )
        Spacer(Modifier.weight(1f))
    }
}
